lib/Actions.js

```diff
@@ -115,7 +172,46 @@ Actions.prototype = {
         }
 
         return vow.all([actionPromise, waitingForPage]).spread(function (result) {
-            return result;
+            return casesPromise || result;
+        });
+    },
+
+    _performCases: function (cases, parentSelector) {
+        debug('handle several cases in parallel %o', cases);
+
+        var wonCase = null;
+        var promises = cases.map(function (actions, caseNumber) {
+            var beginningPromise = this._performAction(actions[0], parentSelector);
+            return actions
+                .slice(1)
+                .reduce(function (promise, action, i, array) {
+                    return promise.then(function () {
+                        if (wonCase !== null && array !== cases[wonCase]) {
+                            return vow.reject('Failed actions chain');
+                        }
+
+                        if (action.trueCase) {
```

Comment:
Imagine we have 3 concurrent action chains:

1. A B C D
2. a b c d
3. 1 2 3 4

On some step of each chain we can understand, that the chain is won the race (needed page is loaded, popup with needed context is shown). It's not necessary the first step, it's different for different chains. 

1. A B **C** D
2. **a** b c d
3. 1 2 3 4

And if such action is happening we must reject all other chains because they can affect parsing process.